server/channels/api4/apitestlib_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api4
+
+import (
+	"os"
+	"testing"
+
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnvironmentVariableHandling(t *testing.T) {
+	// Clean up any existing environment variables before testing
+	originalConsoleLevel := os.Getenv("MM_LOGSETTINGS_CONSOLELEVEL")
+	defer func() {
+		// Restore original environment variables
+		if originalConsoleLevel != "" {
+			os.Setenv("MM_LOGSETTINGS_CONSOLELEVEL", originalConsoleLevel)
```

Comment:
Nothing to change but noting `os.Setenv` is another no-no in terms of parallelizing tests. But in this case it's a bit inevitable given what we are testing. Test should always run serially though.

Reply:
Darn. Yah, good point. Is there a way to ensure serial (like a `t.Serial()` ?)

Reply:
Yah, I meant for the future, to make sure it doesn't get accidentally paralelled

Reply:
I hope you're just joking and you know that I added a comment, and you're making the joke that in the future no-one will read the comment...

Reply:
@streamer45 fixed :)